rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.highlight/50_synthetic_source.yml

```diff
@@ -0,0 +1,314 @@
+
+---
+text multi unified from reanalysis:
+  - do:
+      search:
+        index: test
+        body:
+          query:
+            match_phrase:
+              foo.analyze: "That makes calamity of so long life."
+          highlight:
+            fields:
+              foo.analyze:
+                type: unified
+  - match: { hits.hits.0.highlight.foo\.analyze.0: <em>That</em> <em>makes</em> <em>calamity</em> <em>of</em> <em>so</em> <em>long</em> <em>life</em>. }
+
+---
+text single unified from positions:
+  - do:
+      search:
+        index: test
+        body:
+          query:
+            match_phrase:
+              foo.positions: "the quick brown fox jumped over the lazy dog"
+          highlight:
+            fields:
+              foo.positions:
+                type: unified
+  - match: { hits.hits.0.highlight.foo\.positions.0: <em>the</em> <em>quick</em> <em>brown</em> <em>fox</em> <em>jumped</em> <em>over</em> <em>the</em> <em>lazy</em> <em>dog</em> }
+
+---
+text multi unified from positions:
+  - do:
+      search:
+        index: test
+        body:
+          query:
+            match_phrase:
+              foo.positions: "That makes calamity of so long life."
+          highlight:
+            fields:
+              foo.positions:
+                type: unified
+  - match: { hits.hits.0.highlight.foo\.positions.0: <em>That</em> <em>makes</em> <em>calamity</em> <em>of</em> <em>so</em> <em>long</em> <em>life</em>. }
+
+---
+text single unified from vectors:
+  - do:
+      search:
+        index: test
+        body:
+          query:
+            match_phrase:
+              foo.vectors: "the quick brown fox jumped over the lazy dog"
+          highlight:
+            fields:
+              foo.vectors:
+                type: unified
+  - match: { hits.hits.0.highlight.foo\.vectors.0: <em>the</em> <em>quick</em> <em>brown</em> <em>fox</em> <em>jumped</em> <em>over</em> <em>the</em> <em>lazy</em> <em>dog</em> }
+
+---
+text multi unified from vectors:
+  - do:
+      search:
+        index: test
+        body:
+          query:
+            match_phrase:
+              foo.vectors: "That makes calamity of so long life."
+          highlight:
+            fields:
+              foo.vectors:
+                type: unified
+  - match: { hits.hits.0.highlight.foo\.vectors.0: <em>That</em> <em>makes</em> <em>calamity</em> <em>of</em> <em>so</em> <em>long</em> <em>life</em>. }
+
+---
+text single fvh:
+  - do:
+      search:
+        index: test
+        body:
+          query:
+            match_phrase:
+              foo.vectors: "the quick brown fox jumped over the lazy dog"
+          highlight:
+            fields:
+              foo.vectors:
+                type: fvh
+  - match: { hits.hits.0.highlight.foo\.vectors.0: <em>the quick brown fox jumped over the lazy dog</em> }
+
+---
+text multi fvh:
```

Comment:
Can you add a test that this also occurs when you ask for fragments in order, so that we exercise both FragmentsBuilder implementations?

Reply:
> Can you add a test that this also occurs when you ask for fragments in order, so that we exercise both FragmentsBuilder implementations?

:+1: